### PR TITLE
Deprecate react-unit-test-utils 1/2

### LINKED ISSUE #1748: Deprecate react-unit-test-utils 1/2
Remove use of 'shallow' (react-unit-test-utils) in following components/files:
- src/courseware/course/sequence/Unit/UnitSuspense.test.jsx
- src/courseware/course/sequence/Unit/ContentIFrame.test.jsx

Remove use of 'StrictDict' (react-unit-test-utils) in following components/files:
- src/courseware/course/sequence/Unit/ContentIFrame.jsx
- src/courseware/course/sequence/Unit/constants.ts
- src/courseware/course/sequence/Unit/hooks/useExamAccess.js

Remove use of 'useKeyedState' (react-unit-test-utils) in following components/files:
- src/courseware/course/sequence/Unit/hooks/useExamAccess.js

### PR DESCRIPTION
**Pull request by diana-villalvazo-wgu**
### Description
Remove use of 'shallow' (react-unit-test-utils) in following components/files:
- src/courseware/course/sequence/Unit/UnitSuspense.test.jsx
- src/courseware/course/sequence/Unit/ContentIFrame.test.jsx

Remove use of 'StrictDict' (react-unit-test-utils) in following components/files:
- src/courseware/course/sequence/Unit/ContentIFrame.jsx
- src/courseware/course/sequence/Unit/constants.ts
- src/courseware/course/sequence/Unit/hooks/useExamAccess.js

Remove use of 'useKeyedState' (react-unit-test-utils) in following components/files:
- src/courseware/course/sequence/Unit/hooks/useExamAccess.js

#### Support Information
Closes #1748

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @diana-villalvazo-wgu!


This repository is currently maintained by `@openedx/committers-frontend-app-learning`.


Once you've gone through the following steps feel free to tag them in a comment and let them know that your changes are ready for engineering review.

<details><summary>:radio_button: Get product approval</summary>

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.
</details>

<details><summary>:radio_button: Provide context</summary>

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads
</details>





<details><summary>:radio_button: Get a green build</summary>

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.
</details><details>


---

<details><summary>Where can I find more information?</summary>

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)
</details>

<details><summary>When can I expect my changes to be merged?</summary>

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*
</details>

File: src/courseware/course/sequence/Unit/UnitSuspense.test.jsx [hunk 1]
```diff
@@ -1,33 +1,26 @@
...
   Suspense: 'Suspense',
 }));
 
-jest.mock('../honor-code', () => 'HonorCode');
-jest.mock('../lock-paywall', () => 'LockPaywall');
+jest.mock('../honor-code', () => jest.fn(() => <div>HonorCode</div>));
+jest.mock('../lock-paywall', () => jest.fn(() => <div>LockPaywall</div>));
 jest.mock('@src/generic/model-store', () => ({ useModel: jest.fn() }));
-jest.mock('@src/generic/PageLoading', () => 'PageLoading');
+// jest.mock('@src/generic/PageLoading', () => 'PageLoading');
```
Comment (holaontiveros) [inline src/courseware/course/sequence/Unit/UnitSuspense.test.jsx]: maybe remove this?

Comment (diana-villalvazo-wgu) [reply src/courseware/course/sequence/Unit/UnitSuspense.test.jsx]: done!

**Comment by diana-villalvazo-wgu** [comment]
Hi @openedx/committers-frontend-app-learning can someone help with a review? thanks! 😄

File: src/courseware/course/sequence/Unit/ContentIFrame.test.jsx [hunk 2]
```diff
@@ -89,80 +74,71 @@ describe('ContentIFrame Component', () => {
...
   });
   describe('output', () => {
-    let component;
     describe('if shouldShowContent', () => {
       describe('if not hasLoaded', () => {
         it('displays errorPage if showError', () => {
-          hooks.useIFrameBehavior.mockReturnValueOnce({ ...iframeBehavior, showError: true });
-          el = shallow(<ContentIFrame {...props} />);
-          expect(el.instance.findByType(ErrorPage).length).toEqual(1);
+          hooks.useIFrameBehavior.mockReturnValueOnce({ ...iframeBehavior, showError: true, shouldShowContent: true });
```
Comment (brian-smith-tcril) [inline src/courseware/course/sequence/Unit/ContentIFrame.test.jsx]: It seems this was just missing `shouldShowContent: true` before and this is adding it in to match the test description. Correct?

File: src/courseware/course/sequence/Unit/ContentIFrame.test.jsx [hunk 3]
```diff
@@ -89,80 +74,71 @@ describe('ContentIFrame Component', () => {
...
     describe('if not shouldShowContent', () => {
       it('does not show PageLoading, ErrorPage, or unit-iframe-wrapper', () => {
-        el = shallow(<ContentIFrame {...{ ...props, shouldShowContent: false }} />);
-        expect(el.instance.findByType(PageLoading).length).toEqual(0);
-        expect(el.instance.findByType(ErrorPage).length).toEqual(0);
-        expect(el.instance.findByTestId(testIDs.contentIFrame).length).toEqual(0);
+        render(<ContentIFrame {...{ ...props, shouldShowContent: false }} />);
+        expect(screen.queryByText('PageLoading')).toBeNull();
+        expect(screen.queryByText('ErrorPage')).toBeNull();
+        expect(screen.queryByTitle(props.title)).toBeNull();
```
Comment (brian-smith-tcril) [inline src/courseware/course/sequence/Unit/ContentIFrame.test.jsx:118]: Is there ever a scenario where we wouldn't find the title but could still find the `unit-iframe-wrapper`? Just want to confirm that switching from checking for the `contentIFrame` test id to this still verifies the same thing.

File: src/courseware/course/sequence/Unit/ContentIFrame.test.jsx [hunk 4]
```diff
@@ -89,80 +74,71 @@ describe('ContentIFrame Component', () => {
...
+      render(<ContentIFrame {...props} />);
+      const modal = screen.queryByRole('dialog');
+      expect(modal).toBeNull();
     });
     describe('if modalOptions.isOpen', () => {
-      const testModalOpenAndHandleClose = () => {
-        test('Modal component isOpen, with handleModalClose from hook', () => {
-          expect(component.props.onClose).toEqual(modalIFrameData.handleModalClose);
-        });
-      };
```
Comment (brian-smith-tcril) [inline src/courseware/course/sequence/Unit/ContentIFrame.test.jsx]: It's not clear to me what is replacing this. If it's just being removed without a replacement could you provide some context as to why it isn't needed anymore?

**Review by brian-smith-tcril — COMMENTED** [review]
Overall this looks great!

I left a couple comments in there with some clarifying questions, but nothing major.

Thanks for doing this!

File: src/courseware/course/sequence/Unit/ContentIFrame.test.jsx (hunk 2 above)
Comment (diana-villalvazo-wgu) [reply src/courseware/course/sequence/Unit/ContentIFrame.test.jsx]: oh i think i can delete this, thanks for the heads up

File: src/courseware/course/sequence/Unit/ContentIFrame.test.jsx (hunk 3 above)
Comment (diana-villalvazo-wgu) [reply src/courseware/course/sequence/Unit/ContentIFrame.test.jsx:118]: In the codebase when this component is used, it appends the unit title, only if you don't add a title to a unit that will be applicable 🤔 i can add that use case if needed

File: src/courseware/course/sequence/Unit/ContentIFrame.test.jsx (hunk 2 above)
Comment (diana-villalvazo-wgu) [reply src/courseware/course/sequence/Unit/ContentIFrame.test.jsx]: done!

File: src/courseware/course/sequence/Unit/ContentIFrame.test.jsx (hunk 4 above)
Comment (diana-villalvazo-wgu) [reply src/courseware/course/sequence/Unit/ContentIFrame.test.jsx]: fixed

File: src/courseware/course/sequence/Unit/ContentIFrame.test.jsx (hunk 3 above)
Comment (brian-smith-tcril) [reply src/courseware/course/sequence/Unit/ContentIFrame.test.jsx:118]: I just know that the previous version of this was testing for `testIDs.contentIFrame`, and now it's testing for `props.title`. If there's a scenario where we would have the test id but not the title then that means we've changed what this test is looking for.

Comment (diana-villalvazo-wgu) [reply src/courseware/course/sequence/Unit/ContentIFrame.test.jsx:118]: yeah i made that change in order to follow [Braden's suggestion](https://github.com/openedx/frontend-app-authoring/issues/2075#issuecomment-2940831175) of 
- [Prioritize accessible queries](https://testing-library.com/docs/queries/about#priority) , e.g. use getByRole('button', {name: /submit/i}) instead of getByTestId etc.

I thought get it by title will be the accessible query that fits this use case, but we can bring it back if you think we are missing an edge case here 🤔

Comment (brian-smith-tcril) [reply src/courseware/course/sequence/Unit/ContentIFrame.test.jsx:118]: As a reviewer this is something where having added context would be helpful. Something like:

Previously this was testing for `testIDs.contentIFrame`, which was coming from
https://github.com/openedx/frontend-app-learning/blob/a75c89cd142ca2fa7259efc46f69c2f8dc4133ec/src/courseware/course/sequence/Unit/ContentIFrame.jsx#L26

In the component, the test id is used here

https://github.com/openedx/frontend-app-learning/blob/a75c89cd142ca2fa7259efc46f69c2f8dc4133ec/src/courseware/course/sequence/Unit/ContentIFrame.jsx#L73-L77

this also renders the title, which is preferable to test for to follow [RTL best practices](https://testing-library.com/docs/queries/about/#priority).

Since `title` is a required prop for the `ContentIFrame` component

https://github.com/openedx/frontend-app-learning/blob/a75c89cd142ca2fa7259efc46f69c2f8dc4133ec/src/courseware/course/sequence/Unit/ContentIFrame.jsx#L113

this should still cover all cases that the previous test id based test covered.

Comment (diana-villalvazo-wgu) [reply src/courseware/course/sequence/Unit/ContentIFrame.test.jsx:118]: Sure i will have this in mind on following PRs! Thanks Brian 😄

**Review by brian-smith-tcril — APPROVED** [review]
This LGTM!

One recommendation I have to make the review process smoother would be to provide some more context around the choices you make throughout the process.

The PR description for this provides a summary of the changes that can be seen in the diff, so it covers the *what*. It also includes a general *why* (moving away from deprecated react-unit-test-utils), but nothing specific.

One thing I like to do when making PRs where I've made specific choices is to do a self-review where I provide additional context for some of the changes in the diff. I find this can be very helpful to ensure I have a chance to discuss those specific aspects of the PR with reviewers, or to just preemptively answer questions I think reviewers might have.

Thanks again for pushing this effort forwards! It's great to see the move away from `react-unit-test-utils` have so much momentum!
